samza-core/src/main/java/org/apache/samza/execution/StreamEdge.java

```diff
@@ -111,20 +113,21 @@ boolean isIntermediate() {
   }
 
   Config generateConfig() {
-    Map<String, String> config = new HashMap<>();
+    Map<String, String> newConfig = new HashMap<>();
```

Comment:
Revert rename?

Reply:
'config' override a member field 'config'. Although it is intentional, it makes the code confusing.

Reply:
'streamConfig' would be clearer in that case.

Reply:
will change.